compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1275,10 +1305,19 @@ public void visit(BLangTupleVariable varNode) {
 
         // Create the variable definition statements using the root block stmt created
         createVarDefStmts(varNode, blockStmt, tuple.symbol, null);
-        createRestFieldVarDefStmts(varNode, blockStmt, tuple.symbol);
 
-        // Finally rewrite the populated block statement
-        result = rewrite(blockStmt, env);
+        if (((this.env.scope.owner.tag & SymTag.PACKAGE) == SymTag.PACKAGE)) {
+            // Rest field def stmt virtual variables will be added to init function body, hence change the env
+            SymbolEnv previousEnv = this.env;
+            this.env = this.initFunctionEnv;
+            createRestFieldVarDefStmts(varNode, blockStmt, tuple.symbol);
+            this.env = previousEnv;
+            // If it is a global variable don't rewrite now, will be rewritten later
+            result = blockStmt;
+        } else {
+            createRestFieldVarDefStmts(varNode, blockStmt, tuple.symbol);
+            result = rewrite(blockStmt, env);
+        }
```

Comment:
We can return early and remove the `else` block.

```suggestion
            result = blockStmt;
            return;
        }
        createRestFieldVarDefStmts(varNode, blockStmt, tuple.symbol);
        result = rewrite(blockStmt, env);
```

Reply:
+1